### PR TITLE
RIA-5446 Adding IDAM api and isAdmin field

### PR DESCRIPTION
**Pull request by emma-v1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-5446

### Change description ###
Adding IDAM api and "isAdmin" field to allow us to check what user is currently logged in and show a specific screen to a specific user.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -11,6 +11,23 @@ def type = "java"
 def product = "ia"
 def component = "bail-case-api"
 
+def secrets = [
+  'ia-${env}': [
+    secret('idam-client-id', 'IA_IDAM_CLIENT_ID'),
+    secret('idam-secret', 'IA_IDAM_SECRET'),
```
Comment (neha-aggarw) [inline Jenkinsfile_CNP:17]: Are we using same secret keys as being used in Appeals? Do we need to confirm with someone on this?

Comment (rajeshthuraiyur) [reply Jenkinsfile_CNP:17]: that's a good point, we need to confirm, let me find out

**Review by neha-aggarw — CHANGES_REQUESTED** [review]
Rest of the changes look good. Once we can get confirmation on the keys to use in Authorisation with Idam api, we should be good to go.

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java [hunk 2]
```diff
@@ -5,7 +5,9 @@
 public enum BailCaseFieldDefinition {
     //Example
     APPELLANT_GIVEN_NAMES(
```
Comment (rajeshthuraiyur) [inline src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:7]: Do we need this enum in bail case api?

**Review by rajeshthuraiyur — APPROVED** [review]

File: Jenkinsfile_CNP (hunk 1 above)
Comment (emma-v1) [reply Jenkinsfile_CNP:17]: confirmed with Rajesh that it's ok to proceed for now

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java (hunk 2 above)
Comment (emma-v1) [reply src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:7]: Think it was just an example I'd forgotten to remove - thanks will update

**Review by neha-aggarw — APPROVED** [review]
LGTM.

**Pull request merged**